test/parallel/test-fs-opendir-buffer.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const fs = require('fs');
+const { opendir } = require('fs/promises');
+const path = require('path');
+
+const tmpdir = require('../common/tmpdir');
+
+const testDir = tmpdir.path;
+const files = ['empty', 'files', 'for', 'just', 'testing'];
+
+// Make sure tmp directory is clean
+tmpdir.refresh();
+
+// Create the necessary files
+files.forEach(function(filename) {
+  fs.closeSync(fs.openSync(path.join(testDir, filename), 'w'));
+});
+
+// Opendir from buffer
+{
+  (async () => {
+    const dir = await opendir(Buffer.from(testDir), { encoding: 'buffer' });
+    for await (const dirent of dir)
+      assert.notStrictEqual(typeof dirent.name, null);
```

Comment:
this test can't fail

Reply:
no need of this assertion. removed it, as we only trying to open a directory from buffer.

Reply:
Sure thing. I can remove that function.